### ISSUE TITLE
CAFE5 run is taking forever

### ISSUE THREAD

**Issue by zeynepkurtw** [hahnlab/CAFE5#81]
I've been running CAFE5 with around 12,000 Orthogroups on 4 species by using 32 CPUs. It's been a week and it is still searching for lambda parameters. I haven't played with any values for lambda or gamma, to be able to see the vanilla run with the CAFE. Is that the reason it is taking forever? or, is there a way to estimate the running time?  In the docs, it says 200 Lambda values will be run for 10,000 orthogroups and I could only come to about 70 lambda values far in a week.

Here is how I ran the code:

![Screenshot 2022-08-10 at 09 49 46](https://user-images.githubusercontent.com/34158306/183845387-1303d7e6-b0b6-4acb-8b8d-13ca20f1014e.png)

Thanks in advance!

**Comment by benfulton** [comment]
No, it won't take a week. It's clearly locked up somewhere - likely in a thread race issue. Try it with just 8 threads to try to minimize that possibility. I would allow an hour per 1000 groups in that case, no more.

**Comment by zeynepkurtw** [comment]
I see but still, it didn't help with my case. I am suspicious of my tree file as I've used the direct output from Orthofinder without trying to make it an ultrametric tree. I suppose CAFE5 can handle that, right?

 My tree file looks like this:

`(kbiala:0.476202,((spiro:0.575389,(HIN:0.33093,trepo:0.374071)0.867084:0.201903)0.809226:0.198319,(muris:0.401597,wb:0.413952)0.97889:0.327973)1:0.476202);`

or should I fix the lambda value to 1 with -l parameter?

**Comment by hahnlab-user** [comment]
Your tree should be ultrametric, though the program will not throw an error if it's not.

**Comment by zeynepkurtw** [comment]
Okay, I've made an ultrametric tree and used only 8 cores but still, Cafe has been running for 17 hours with 14 000 orthogroups.

Now my ultrametric tree looks like this:

`(kbiala:1.2505,((spiro:0.625248,(HIN:0.312624,trepo:0.312624)1:0.312624)1:0.312624,(muris:0.468936,wb:0.468936)1:0.468936)1:0.312624);
`

**Comment by benfulton** [comment]
Roughly how many lambdas has it generated, and how different are the lambdas near the end? If they aren't that different, there are parameters that would have it stop without trying to get the high precision that is the default.

**Comment by zeynepkurtw** [comment]
So far it has generated around 45 lambda values and doesn't seem to be approaching any particular value. Below are the lambda values near the end:

![lambda](https://user-images.githubusercontent.com/34158306/184530287-3cfb7fd9-556e-4b32-a907-6a9840e54ad4.png)

Why do you think that is the case? I have one species with a highly duplicated genome and that might be the reason. In the CAFE tutorial, it is mentioned that filtration needs to be performed by a script:

_"Gene families that have large gene copy number variance can cause parameter estimates to be non-informative. You can remove gene families with large variance from your dataset, but we found that putting aside the gene families in which one or more species have ≥ 100 gene copies does the trick."_

Is the filtration process already involved in the CAFE5 script or do I have to run the script priorly?
 
Actually, I'm not sure which lambda value I should choose. Isn't that supposed to be calculated by the software? or Can I just choose a lambda value by try and error?

**Comment by hahnlab-user** [comment]
Yes, the fact that you're getting "inf" likelihoods from the very beginning means that no likelihood can be calculated. You may have to further filter your data, before running CAFE.
